modules/apps/portal-workflow/portal-workflow-api/src/main/java/com/liferay/portal/workflow/definition/groovy/script/use/WorkflowDefinitionGroovyScriptUseSourceURLFactory.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.workflow.definition.groovy.script.use;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.GroupConstants;
+import com.liferay.portal.kernel.util.HttpComponentsUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.util.PropsValues;
+import com.liferay.portal.workflow.constants.WorkflowPortletKeys;
+import com.liferay.portal.workflow.constants.WorkflowWebKeys;
+import com.liferay.portal.workflow.portlet.tab.WorkflowPortletTabServiceTracker;
+
+import javax.portlet.PortletMode;
+import javax.portlet.WindowState;
+
+/**
+ * @author Feliphe Marinho
+ */
+public class WorkflowDefinitionGroovyScriptUseSourceURLFactory {
+
+	public static String create(
+			Company company, Portal portal, String workflowDefinitionName,
+			int workflowDefinitionVersion,
+			WorkflowPortletTabServiceTracker workflowPortletTabServiceTracker)
+		throws Exception {
+
+		if (workflowPortletTabServiceTracker.contains(
+				WorkflowWebKeys.WORKFLOW_TAB_DEFINITION)) {
+
+			String url = _getBaseURL(
+				company, WorkflowPortletKeys.KALEO_DESIGNER);
+
+			String namespace = portal.getPortletNamespace(
+				WorkflowPortletKeys.KALEO_DESIGNER);
+
+			url = HttpComponentsUtil.addParameter(
+				url, namespace + "clearSessionMessage", true);
+			url = HttpComponentsUtil.addParameter(
+				url, namespace + "draftVersion",
+				workflowDefinitionVersion + ".0");
+			url = HttpComponentsUtil.addParameter(
+				url, namespace + "mvcPath",
+				"/designer/edit_workflow_definition.jsp");
+			url = HttpComponentsUtil.addParameter(
+				url, namespace + "name", workflowDefinitionName);
+
+			return HttpComponentsUtil.addParameter(
+				url, namespace + "redirect",
+				_getBaseURL(
+					company, WorkflowPortletKeys.CONTROL_PANEL_WORKFLOW));
+		}
+
+		String url = _getBaseURL(
+			company, WorkflowPortletKeys.CONTROL_PANEL_WORKFLOW);
+
+		String namespace = portal.getPortletNamespace(
+			WorkflowPortletKeys.CONTROL_PANEL_WORKFLOW);
+
+		url = HttpComponentsUtil.addParameter(
+			url, namespace + "mvcPath",
+			"/definition/edit_workflow_definition.jsp");
+
+		return HttpComponentsUtil.addParameter(
+			url, namespace + "redirect",
+			HttpComponentsUtil.addParameter(
+				_getBaseURL(
```

Comment:
Same thing, you are invoking `_getBaseURL(company, WorkflowPortletKeys.CONTROL_PANEL_WORKFLOW)` twice here, maybe you can reuse the value from the first invoke to avoid more than one call.

Reply:
Done [here](https://github.com/liferay-workflow/liferay-portal/pull/1103/commits/a6196db56ed019ce0b49d8c6d06d90a932f51bbd)